### ISSUE TITLE
FR: Add `metric` option to `CoreSet`.

### ISSUE THREAD

**Issue by mherde** [scikit-activeml/scikit-activeml#516]
### Describe the workflow you want to enable

The query strategy should allow for modification of the metric.

### Describe your proposed solution

Add `metric` and `metric_dict` as initialization parameters to be passed to `sklearn.metrics.pairwise_distances_argmin_min`.
